docs/pages/reference/access-controls/roles.mdx

```diff
@@ -212,6 +212,12 @@ spec:
       contains(user.spec.traits["teams"], labels["team"])
 ```
 
+<Admonition type="note">
+Usage of label expressions must be the only label logic in roles
+assigned to an identity. For example you cannot have `node_labels`
+and `node_labels_expressions` in the same role or from separate roles.
+</Admonition>
+
```

Comment:
this is not correct, a couple paragraphs down i have written

> Typically you will only want to use one of `<kind>_labels` or
`<kind>_labels_expression` in a single role, but you are allowed to specify
both.
If you have both in a deny rule, the matching is greedy, if either one matches
access will be denied.
In an allow rule, the matching is not greedy, if both are set they both have to
match for access to be allowed.

Reply:
When I tested having both only the `node_labels` works. The labels expr is ignored.

Reply:
If I put this will not see any resources. Individually will see the resources if one or the other is removed.

```yaml
  allow:
    logins:
    - '{{internal.logins}}'
    - ubuntu
    node_labels:
      aws/env: staging
    node_labels_expression: labels["aws/env"] == "dev"
```

Reply:
> In an allow rule, the matching is not greedy, if both are set they both have to
match for access to be allowed.

it's AND not OR, since the aws/env label can't match both staging and dev at the same time, you won't see anything

Reply:
I see thanks!